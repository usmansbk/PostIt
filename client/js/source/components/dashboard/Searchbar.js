import React from 'react';
import Icon from '../common/Icon';

export default function Searchbar ({onClick}) {
  return (
    <div className='navbar-fixed nav-wrapper'>
      <nav>
        <div className='input-field'>
          <input id='searchbar' type='search' className='center-align grey lighten-3' placeholder='Search PostIt' />
          <label className='label-icon' htmlFor='searchbar'><Icon>search</Icon></label>
          <Icon name='cancel-search' onClick={onClick}>close</Icon>
        </div>
      </nav>
    </div>
  )
}
