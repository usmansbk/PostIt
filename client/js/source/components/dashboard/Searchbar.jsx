import React from 'react';
import Icon from '../common/Icon.jsx';

export default function Searchbar (props) {
  return (
    <div className='navbar-fixed'>
      <nav>
        <div className='input-field'>
          <input id='searchbar' type='search' className='center-align grey lighten-3' placeholder='Search PostIt' />
          <label className='label-icon' htmlFor='searchbar'><Icon>search</Icon></label>
          <Icon name='cancel-search' onClick={props.onClick}>close</Icon>
        </div>
      </nav>
    </div>
  )
}
