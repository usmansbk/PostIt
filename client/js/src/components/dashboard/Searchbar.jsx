import React from 'react';
import Icon from '../common/Icon';

export default ({onClick, onChange, children}) => {
  return (
    <div className='navbar-fixed nav-wrapper'>
      <nav>
        <div className='input-field'>
          <input id='searchbar' type='search' className='center-align grey lighten-3' placeholder='Search PostIt' onChange={onChange}/>
          <label className='label-icon grey-text' htmlFor='searchbar'><Icon>search</Icon></label>
          <Icon name='cancel-search' onClick={onClick}>close</Icon>
        </div>
        {children}
      </nav>
    </div>
  )
}
