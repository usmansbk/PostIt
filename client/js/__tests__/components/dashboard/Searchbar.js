import React from 'react';
import Enzyme, { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Searchbar from '../../../src/components/dashboard/Searchbar';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    onClick: jest.fn(),
  }

  const mockEvent = {};
  mockEvent.preventDefault = jest.fn();

  const wrapper = mount(<Searchbar {...props } />)

  return {
    props,
    wrapper,
    mockEvent
  }
}

describe('Searchbar', () => {
  it('should render component', () => {
    const component = renderer.create(<Searchbar />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should call onClick', () => {
    const { wrapper, props, mockEvent } = setup();
    const icon = wrapper.find('i[name="cancel-search"]');
    icon.simulate('click', mockEvent);
    expect(props.onClick).toBeCalled();
  })
})