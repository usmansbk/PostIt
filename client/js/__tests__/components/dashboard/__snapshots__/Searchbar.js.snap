// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar should render component 1`] = `
<div
  className="navbar-fixed nav-wrapper"
>
  <nav>
    <div
      className="input-field"
    >
      <input
        className="center-align grey lighten-3"
        id="searchbar"
        onChange={undefined}
        placeholder="Search PostIt"
        type="search"
      />
      <label
        className="label-icon grey-text"
        htmlFor="searchbar"
      >
        <i
          className="material-icons undefined"
        >
          search
        </i>
      </label>
      <i
        className="material-icons undefined"
        name="cancel-search"
        onClick={undefined}
      >
        close
      </i>
    </div>
  </nav>
</div>
`;
